kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -246,6 +246,9 @@ private[kyuubi] class EngineRef(
               ms.incCount(MetricRegistry.name(ENGINE_FAIL, error.getClass.getSimpleName))
             }
             throw error
+          } else {
+            acquiredPermit = false
+            startupProcessSemaphore.foreach(_.release())
```

Comment:
not a big deal, let's invert the `if else` to avoid anti-condition to gain clearer code

```
if (exitValue.contains(0)) {
  startupProcessSemaphore.foreach(_.release())
  acquiredPermit = false
} else {
  ...
}
```

Reply:
I got it